### ISSUE TITLE
NEMA non-barrier cross transitions should skip Y->R transition if same phase. 

### ISSUE THREAD

**Issue by mschrader15** [eclipse/sumo#9987]
As it stands NEMA goes through a Y -> R -> G transition on 6 to go from 1+6  to 2+6. 

 Instead, 6 should stay green.

**Current behavior**

https://github.com/eclipse/sumo/blob/1563c460c657077c7261845c34f87b3b6fa8668b/tests/sumo/basic/tls/NEMA/fixForceOff/ramp_low_ffO_true_minRecall/tls_state_switch.sumo#L41-L44

**Expected behavior**
```
 <tlsState time="11.00" id="2881" programID="01" phase="1" state="srrrGGrrGGG" name="2+6"/>
 <tlsState time="80.00" id="2881" programID="01" phase="0" state="srrrGGrryyy" name="2+6"/>
 <tlsState time="84.00" id="2881" programID="01" phase="1" state="srrrGGrrrrr" name="2+6"/>
 <tlsState time="85.00" id="2881" programID="01" phase="0" state="srrrGGGGrrr" name="1+6"/>
```

**Comment by mschrader15** [comment]
Important to note (partly for myself) that #9987 was present before #9970

**Comment by qichaow** [comment]
@mschrader15 wait. Something is wrong. Phase 6 should stay green when going from 1+6 to 2+6 in the existing code at least the last time I looked. Let's check before moving forward.

**Comment by mschrader15** [comment]
@qichaow I rebuilt SUMO from [325db1c](https://github.com/eclipse/sumo/blob/325db1c8ea0311b9d0f8c9c9299d595c4cbbcff6/src/microsim/traffic_lights/NEMAController.cpp) which is before I ever touched the `NEMAController.cpp` 😆 

I am still seeing the issue I describer here. Below is the phase tracker.
![image](https://user-images.githubusercontent.com/57813331/150136269-50f84401-2347-44d4-8963-62ad10c1b331.png)

**Comment by mschrader15** [comment]
@qichaow I did some more digging and have an explanation:

Your code (commits [325db1c](https://github.com/eclipse/sumo/blob/325db1c8ea0311b9d0f8c9c9299d595c4cbbcff6/src/microsim/traffic_lights/NEMAController.cpp) and before) supported holding 6 green when going from 1+6 -> 2+6. 

The example above is going from 2+6 -> 1+6. That is technically all the way around ring 1 (2 -> 3 -> 4 -> 1). Your code forced both phases to go red if there was a "barrier cross". While that maybe the behavior of some traffic lights, I have seen in testing with our SIL controller that in **both free and coordinated mode** going from 2+6 -> 1+6 should not require 6 to go red.

If that is not the general case, we can make https://github.com/eclipse/sumo/blob/597e9bba6d32b71c78693bace97f8a323278c916/src/microsim/traffic_lights/NEMAController.cpp#L109 default to false, which will cause #9988 to behave like your original code

**Comment by qichaow** [comment]
That makes sense.

**Comment by mschrader15** [comment]
@qichaow #9988 implements a solution

https://github.com/eclipse/sumo/blob/cfd901056698f55e1b7f4531fc9610393fcbeeb8/tests/sumo/basic/tls/NEMA/fixForceOff/ramp_low_ffO_true_minRecall/tls_state_switch.sumo#L41-L44

**Comment by qichaow** [comment]
> @qichaow Anything still in the way of merging #9988 ?

Not really. It would still be good to fix the test timing though. It won't break the current code but will cause an error when we merge the error handling PR. These kind of the timing issue actually exists in the field but is not a good practice. We can also merge it now and fix the timing later.

**Comment by qichaow** [comment]
Cool.

**Comment by mschrader15** [comment]
@qichaow See https://github.com/eclipse/sumo/issues/9709#issuecomment-1016538037 haha.

For the test cases that I have created I have been using timings from the field.

**Comment by qichaow** [comment]
@namdre I don't have any concern about merging #9988. I will just throw a warning instead since I did see that in the field although it's not a good practice. 

@mschrader15 @namdre 
sorry I have been responding by email and GitHub on my phone. Some responses may not match the latest comments.

**Comment by mschrader15** [comment]
@qichaow no problem!

**Comment by namdre** [comment]
> Not really. It would still be good to fix the test timing though. It won't break the current code but will cause an error when we merge the error handling PR. 

That's a good thing. We need tests for the error handling as well.
